File: steakfrye-cs50-2018-x-cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //declare float variables
    float change;
    int coins, change2;

    coins = 0;

    //ask user for change
    do
    {
        change = get_float("Change owed: ");
    }

    while (change < 0);
    {
        change2 = round(change * 100);

        //Calculate quarters
        while (change2 >= 25)
        {
            coins = (coins + 1);
            change2 = (change2 - 25);
        }
        //Calculate dimes
        while (change2 >= 10)
        {
            coins = (coins + 1);
            change2 = (change2 - 10);
        }
        //Calculate nickels
        while (change2 >= 5)
        {
            coins = (coins + 1);
            change2 = (change2 - 5);
        }
        //Calculate pennies
        while (change2 >= 1)
        {
            coins = (coins + 1);
            change2 = (change2 - 1);
        }

        printf("%i Coins used.\n", coins);
    }
}

File: pset6/cash.py
#returns amount of change owed with fewest coins possible used
from cs50 import get_float

#prompt user for change until given
while True:
    change = get_float("Change owed: ")
    if change > 0:
        break
#convert dollars to cents
change *= 100

#declare coins
quarter = 25
dime = 10
nickel = 5
penny = 1
coins = 0

#calculate how many coins will need to be used
while change >= quarter:
    change -= quarter
    coins += 1
while change >= dime:
    change -= dime
    coins += 1
while change >= nickel:
    change -= nickel
    coins += 1
while change >= penny:
    change -= penny
    coins += 1

print(f"Coins used: {coins}")